### DISCUSSION TITLE
Allow defining `maxFacetHits` independently from `showMoreLimit` in refinementLists that are searchable

### DISCUSSION THREAD

**Discussion by yonguelink** [algolia/instantsearch#4500] [General]
A refinementList that is `searchable` and has `showMore: true` & `showMoreLimit` set to > 100 & you have clicked the `Show more` button, the search box does not work. That's because the `maxFacetHits` is [limited to 100 for performance reasons](https://www.algolia.com/doc/api-reference/api-parameters/maxFacetHits/#usage-notes:~:text=For%20performance%20reasons%2C%20the%20maximum%20allowed,range%20%5B1%2C%20100%5D%20will%20be%20rejected.).

## Steps to reproduce the behavior

1. Go to https://codesandbox.io/s/infallible-aryabhata-l1w8x
1. Open your browser console (not the one in codesandbox)
2. Click on `Show more`
3. Type anything in the search box
4. See an error in the console

```js
{
  debugData: [{…}]
  message: "Value too large for \"maxFacetHits\" parameter, expected integer between 1 and 100"
  name: "AlgoliaSearchError"
  statusCode: 400
  stack: "AlgoliaSearchError: Value too large for \"maxFacetHits\" parameter, expected integer between 1 and 100↵    at c (https://cdn.jsdelivr.net/algoliasearch/3.32.0/algoliasearchLite.min.js:2:21121)↵    at async Promise.all (index 0)"
}
```

**Live reproduction:**

https://codesandbox.io/s/infallible-aryabhata-l1w8x

**🖥 Screenshots**

![image](https://user-images.githubusercontent.com/9469187/91874331-19085b80-ec48-11ea-9043-9e611c5eb699.png)

![image](https://user-images.githubusercontent.com/9469187/91874361-24f41d80-ec48-11ea-9232-76309b5c46d4.png)

## Feature suggestion

I think it'd be a good option to be able to define `maxFacetHits` independently from the `showMoreLimit` in refinement lists. That way we'd be able to show however many values in one when no search is applied and still respect the API for `maxFacetHits`.

**Comment by Haroenv** [comment]
It may make sense to limit https://github.com/algolia/instantsearch.js/blob/master/src/connectors/refinement-list/connectRefinementList.js#L252 to 100 automatically so no error gets displayed, so that would become:


```js
helper
  .searchForFacetValues(attribute, query, Math.max(getLimit(isShowingMore), 100), tags)
```

We probably want to add a `warning` in dev only as well when `getLimit(isShowingMore) > 100`

Would you be willing to make a pull request limiting the maxFacetHits?

In the mean time what you would be able to do (not so ideal, but possible) is create a custom search client ([docs](https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/)) and transform the parameter `maxFacetHits` to be maximum 100

What do you think?

**Reply by yonguelink** [reply]
This seems like a decent work around. It would avoid crashing and not returning anything.

For the problem we hit, we simply reduced our `showMoreLimit` to 100.

I'll check if I can make a PR without too much trouble.
